Weather Man/weather_parser.py

```diff
@@ -0,0 +1,111 @@
+"""
+weather_parser.py
+
+This module defines the WeatherParser class, which parses weather data files year-wise and month-wise for Lahore,
+Pakistan.
+
+Classes:
+- WeatherParser: A class to parse weather data files year-wise and month-wise from a specified directory.
+
+"""
+
+import os
+from constants import month_map
+from datetime import datetime
+from weather_data import WeatherData
+import pandas as pd
+
+
+def parse_file(file_path):
+    """
+    Parses a CSV file containing weather data and returns a WeatherData object.
+
+    Args:
+    - file_path (str): Path to the CSV file containing weather data.
+
+    Returns:
+    - Object: A WeatherData object containing parsed weather readings.
+    """
+    readings = []
+    data = pd.read_csv(file_path)
+    if 'PKST' in data.columns:
+        data.rename(columns={'PKST': 'PKT'}, inplace=True)
+    data = data[
+        ["PKT", "Max TemperatureC", "Mean TemperatureC", "Min TemperatureC", "Max Humidity", " Mean Humidity",
+         " Min Humidity"]]
+    data.rename(columns={"PKT": "date", " Mean Humidity": "Mean Humidity", " Min Humidity": "Min Humidity"},
+                inplace=True)
+    data.dropna(inplace=True)
```

Comment:
We dont need to rename, Use other ways to strip the extra space. Also PKT/Pkst is not required to be renamed. make it flexible for both